pkg/build/build.go

```diff
@@ -0,0 +1,85 @@
+package build
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/suse-edge/edge-image-builder/pkg/config"
+)
+
+func Build(imageConfig *config.ImageConfig, buildConfig *config.BuildConfig) error {
+	err := prepareBuildDir(buildConfig)
+	if err != nil {
+		return err
+	}
+
+	err = ConfigureMessage(buildConfig)
+	if err != nil {
+		return err
+	}
+
+	err = cleanUpBuildDir(buildConfig)
+	return err
+}
+
+//go:embed scripts/script_base.sh
+var scriptBase string
+
+func prepareBuildDir(buildConfig *config.BuildConfig) error {
+
+	/* Combustion works by creating a volume with a subdirectory named "combustion"
+	   and a file named "script". This function builds out that structure and updates
+	   the BuildConfig so that the other functions can populate it as necessary.
+	*/
+
+	if buildConfig.BuildTempDir == "" {
+		tmpDir, err := os.MkdirTemp("", "eib-")
+		if err != nil {
+			return err
+		}
+		buildConfig.BuildTempDir = tmpDir
+	}
```

Comment:
I'm a little confused here. Generally, this is part of the config initialization logic so we should not be concerned with it in the `build` package. Do we expect that if a `BuildTempDir` is provided, then it exists?

Reply:
Yes, exactly. My next patch (the one I was talking about with regard to rebases in Slack) is actually going to allow the user to indicate the build directory (I may change the name of this depending on how it evolves). The idea was that this function takes care of any sort of operations related to initializing that build directory, so creating subdirs, etc. I considered the optional-ness of it neededing to be specified as something the builder itself is responsible for resolving, because I don't think that's logic that should exist in the CLI itself (it shouldn't care about how to resolve optional values, simply that it can or cannot accept them) and it didn't make sense to live in the config itself.

Reply:
Going to leave as is for now, but call me out on this in the future once we're reading in the user build dir.